apps/node/server.js

```diff
@@ -0,0 +1,160 @@
+const express = require("express");
+const axios = require("axios");
+const dotenv = require("dotenv").config();
+const crypto = require("crypto");
+const qs = require("qs");
+
+const { getState, setState } = require("./dal");
+
+const ATTENTIVE_API_URL = "https://api-devel.attentivemobile.com/v1";
+const ACCESS_TOKEN_ENDPOINT =
+  "https://api-devel.attentivemobile.com/v1/authorization-codes/tokens";
+
+const CLIENT_ID = dotenv.parsed.CLIENT_ID; // Your client id
+const CLIENT_SECRET = dotenv.parsed.CLIENT_SECRET; // Your secret
+const REDIRECT_URI = dotenv.parsed.REDIRECT_URI; // Your redirect uri
+const PORT = 3002;
+
+let app = express();
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  try {
+    res.json({ message: "Your Application is running successfully!" });
+  } catch (ex) {
+    console.error("unexpected error", ex);
+    res.status(500).json({ message: "Unexpected error" });
+  }
+});
+
+app.get("/install", async (req, res) => {
+  try {
+    // Generates a random string containing numbers and letters
+    const state = crypto.randomBytes(5).toString("hex");
+
+    // List of scopes that your app needs
+    const scopes = ["ecommerce:write", "events:write", "subscriptions:write"];
+
+    setState(state, {});
+
+    // build redirect url and return 302
+    const redirect_url =
+      `https://ui-devel.attentivemobile.com/integrations/oauth-install?client_id=${CLIENT_ID}` +
+      `&redirect_uri=${REDIRECT_URI}` +
+      `&state=${state}`;
+    res.status(302).redirect(redirect_url);
+  } catch (ex) {
+    console.error("unexpected error", ex);
+    res.status(500).json({ message: "Unexpected error" });
+  }
+});
+
+app.get("/callback", async (req, res) => {
+  try {
+    const authorization_code = req.query.code;
```

Comment:
Style wise I'd destructure here `const { code, state } = req.query`, but up to you.

Reply:
I added it, looks more professional